cli/cmd/generate.go

```diff
@@ -248,16 +261,67 @@ func patchTargets(paths []string, logger *slog.Logger) error {
 			return fmt.Errorf("failed to unmarshal %s: %w", path, err)
 		}
 
-		var changed bool
-		replaceRuntimeClassName := runtimeClassNamePatcher(&changed)
+		for i, resource := range kubeObjs {
+			deploy, ok := resource.(*applyappsv1.StatefulSetApplyConfiguration)
+			if ok && deploy.Spec.Template.Annotations[contrastRoleAnnotationKey] == "coordinator" {
+				continue
+			}
+			resource, err = kuberesource.AddInitializer(resource, kuberesource.Initializer())
+			if err != nil {
+				return err
+			}
+			kubeObjs[i] = resource
+		}
+
+		logger.Debug("Updating resources in yaml file", "path", path)
+		modifiedData, err := kuberesource.EncodeResources(kubeObjs...)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(path, modifiedData, os.ModePerm); err != nil {
+			return fmt.Errorf("failed to write %s: %w", path, err)
+		}
+	}
+	return nil
+}
+
+func patchTargets(paths []string, imageReplacementsFile string, logger *slog.Logger) error {
+	var replacements map[string]string
+	var err error
+	if imageReplacementsFile != "" {
+		f, err := os.Open(imageReplacementsFile)
+		if err != nil {
+			return fmt.Errorf("could not open image replacements file %s: %w", imageReplacementsFile, err)
```

Comment:
```suggestion
			return fmt.Errorf("opening image replacements file %s: %w", imageReplacementsFile, err)
```
In general, we don't wrap erros with `failed to ...` or `could not ...`, as errors are usually wrapped multiple times and those phrases will stack up:

```
Error: failed to foo: failed to bar: could not open image replacements file "": could not read file: file does not exist
```
Only the inner error (like error creates with `errors.New`) should use failed.